### PR TITLE
Implement reposting from vk.com to telegram

### PR DESCRIPTION
**Pull request by Wasper256**
- Download text and pictures from **Vk** to **Telegram**

### CODE REVIEW

File: vk.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if(current_id - prev_id != 0):
+        text = text + or_link
+        rp = str(resp)
+        rp = rp.split()
+        c = rp.count("'src_big':")
+        if c > 0:
+            pos = rp.index("'src_big':")  # getting image url
+            url = (rp[pos + 1])
+            url = re.sub(r"\'|\,", '', url)
+            print(text + url)
```
Comment (ogroleg) [inline vk.py]: [logging](https://docs.python.org/3/library/logging.html)?

**Comment by cleac** [comment]
@gromoleg :+1:

File: telebot.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import requests
+import constants
+import urllib
+import io
+
+
+def tbotxt(text):
```
Comment (cleac) [inline telebot.py]: Name of this function and function `tbotimg` is kinda tautology. Maybe it's better to rename them to `send_text` and `send_picture`?

File: vk.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    response = final_data['response'][1:2]
+    for resp in response:
+        current_id = resp['id']  # getting an id
+        text = resp['text']  # getting txt
+    # getting previous id and writing new id
+    if os.path.exists('last_id.txt'):
+        with open('last_id.txt') as f:
+            prev_id = f.read()
+    else:
+        prev_id = '0'
```
Comment (cleac) [inline vk.py]: Maybe it's much better to make like this
```python
prev_id = 0
if os.path.exists('last_id.txt'):
    with open('last_id.txt') as f:
        prev_id = int(f.read())
```

Perhaps, did I told you that you should use configs here instead this strange file?

File: vk.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    or_link = "\nOriginal link https://vk.com/wall{0}_{1}".format(vk_id, current_id)
+    current_id = int(current_id)  # and now it must be integer
+    prev_id = int(prev_id)
+    if(current_id - prev_id != 0):
+        text = text + or_link
+        rp = str(resp)
+        rp = rp.split()
+        c = rp.count("'src_big':")
+        if c > 0:
+            pos = rp.index("'src_big':")  # getting image url
```
Comment (cleac) [inline vk.py]: Hadn't I told you that it is not ok to do this? Isn't it easier to understand and faster to get picture like checking whether there is a field in `dict`?

Comment (cleac) [inline vk.py]: Can you type here, what do you do from line _34_ till line _38_?

Comment (Wasper256) [inline vk.py]: 34  doing string from dict.
35  split string to list
36-37 testing do we have image in post or not
38 getting position of image url

Comment (cleac) [inline vk.py]: I read that :-) Explain, why or what for do you do it?

Comment (Wasper256) [inline vk.py]: in main dictionary we have tab "attachments" that return string(not dictionary) with many urls of one picture(in diferent sizes). that is why i must split that on parts and get position of "src_big:" next point in list will be our urll.

Comment (cleac) [inline vk.py]: While you are looking for `'src_big':` in it, it should be `json` formatted. Why not parse it with json and operate with it, like a `dict`?

Comment (Wasper256) [inline vk.py]: done

File: main.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import threading
+import vk2telegram
+# repeater for our script. check for updates every 5 min
```
Comment (cleac) [inline main.py]: There is pyDoc for this staff. It looks like this in code:

``` python
"""Module documentation: what it is, what it does"""

def fun():
    """Function documentation: what it does"""
    pass
```

File: telebot.py (hunk 2 above)
Comment (cleac) [inline telebot.py]: Repeat

> Name of this function and function "tbotimg" is kinda tautology. Maybe it's better to rename them to "send_text" and "send_picture"?

File: telebot.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    params = urllib.parse.quote(text, safe='')  # formatting text for url
+    url_send = "https://api.telegram.org/bot{0}/sendmessage?chat_id={1}&text={2}".format(constants.token, constants.chat_id, params)
+    urllib.request.urlopen(url_send)
+
+
+def tbotimg(url):
+    data = {'chat_id': constants.chat_id}
+    bnr = urllib.request.urlopen(url).read()  # binary response
+    raw = io.BytesIO(bnr)  # raw image
+    raw.name = "bullshit.jpg"  # setting name for image. must have for telegram
```
Comment (cleac) [inline telebot.py]: `raw.name = "bullshit.jpg"` - that's impolite

File: vk.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import os
+import json
+import urllib.request
+import constants
+
+
+def vk_get():
```
Comment (cleac) [inline vk.py]: `get_post(owner_id)` - and pass owner id as function argument

File: vk.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    adr = 'https://api.vk.com/method/wall.get?owner_id={0}&count=1'.format(vk_id)
+    data = urllib.request.urlopen(adr)
+    decoded_response = data.read().decode()
+    final_data = json.loads(decoded_response)
+    # getting last post response
+    response = final_data['response'][1:2]
+    for resp in response:
+        current_id = resp['id']  # getting an id
+        text = resp['text']  # getting txt
+        has_big_pic = 0
```
Comment (cleac) [inline vk.py]: `False`

File: vk.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        if 'src_big' in resp:
+            has_big_pic = 1
```
Comment (cleac) [inline vk.py]: `True`

File: vk.py (hunk 3 above)
Comment (cleac) [inline vk.py]: Again - do you read what I'm telling to you? I've written about this three days ago!!! :angry:

File: vk.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    or_link = "\nOriginal link https://vk.com/wall{0}_{1}".format(vk_id, current_id)
+    current_id = int(current_id)  # and now it must be integer
+    prev_id = int(prev_id)
+    if(current_id - prev_id != 0):
+        text = text + or_link
+        if has_big_pic == 1:
+            return text, url
+        return text, False
+    else:
+        return False, False
```
Comment (cleac) [inline vk.py]: I've told you that it is shitty to pass parameters of different types in one function. Return `bool` **OR** `str`, not both.
Make it 

``` python
         text += or_link
         if has_big_pic:
             return text, url
         return text, ""  # or text, None
     else:
         return "", ""  # or None, None
```

File: vk.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+            prev_id = f.read()
+    else:
+        prev_id = '0'
+    with open('last_id.txt', 'w') as f:
+        current_id = str(current_id)  # must be str for writing/creating link
+        f.write(str(current_id))
+    or_link = "\nOriginal link https://vk.com/wall{0}_{1}".format(vk_id, current_id)
+    current_id = int(current_id)  # and now it must be integer
+    prev_id = int(prev_id)
+    if(current_id - prev_id != 0):
```
Comment (cleac) [inline vk.py]: `current_id != prev_id` - don't do redundant math

File: telebot.py [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import requests
+import constants
+import urllib
+import io
+
+
+def tbotxt(text):
+    params = urllib.parse.quote(text, safe='')  # formatting text for url
+    url_send = "https://api.telegram.org/bot{0}/sendmessage?chat_id={1}&text={2}".format(constants.token, constants.chat_id, params)
```
Comment (cleac) [inline telebot.py]: 1 kilometer str - looks gorgeous _(sarcasm)_

**Comment by Wasper256** [comment]
fixed all.

File: telebot.py [hunk 13]
```diff
@@ -1,19 +1,29 @@
...
 
 def tbotxt(text):
+    """Sending text to telegram."""
     params = urllib.parse.quote(text, safe='')  # formatting text for url
-    url_send = "https://api.telegram.org/bot{0}/sendmessage?chat_id={1}&text={2}".format(constants.token, constants.chat_id, params)
+    url_send = (
+        "https://api.telegram.org/bot{0}" +
+        "/sendmessage?chat_id={1}&text={2}").format(
+        constants.token, constants.chat_id, params,
+    )
```
Comment (cleac) [inline telebot.py:17]: It's nice, but I've found much more beautiful way to express it 

``` python
url_send = "https://api.telegram.org/bot{}/sendmessage?{}".format(
    constants.token,
    urlencode({
        "chat_id": constants.chat_id,
        "text": params
    })
)
```

_urlencode is inside_ `urllib.parser`

File: telebot.py [hunk 14]
```diff
@@ -1,19 +1,29 @@
...
     bnr = urllib.request.urlopen(url).read()  # binary response
     raw = io.BytesIO(bnr)  # raw image
-    raw.name = "bullshit.jpg"  # setting name for image. must have for telegram
+    raw.name = "image.jpg"  # setting name for image. must have for telegram
     files = {'photo': io.BufferedReader(raw)}
-    requests.post("https://api.telegram.org/bot{0}".format(constants.token) + '/sendPhoto', data=data, files=files)
+    requests.post(
+        "https://api.telegram.org/bot{0}".format(constants.token) +
+        "/sendPhoto", data=data, files=files
+    )
```
Comment (cleac) [inline telebot.py:31]: Now here it'd work a bit slow. Here you should make 

``` python
requests.post(
    "https:///api.telegram.org/bot{}/sendPhoto".format(constants.token),
    data=data, files=files
)
```

File: vk2telegram.py [hunk 15]
```diff
@@ -1,9 +1,13 @@
...
 import vk
 import telebot
+import constants
 
 
 def vk2telegram():
-    response = vk.vk_get()
+    """Running scrips for getting and posting posts."""
+    vk_id = constants.vk_id
+    response = vk.vk_get(vk_id)
```
Comment (cleac) [inline vk2telegram.py]: It's a tautology, isn't it? 

```
vk.vk_get
```

File: vk.py [hunk 16]
```diff
@@ -1,13 +1,15 @@
...
 
 
-def vk_get():
-    # get data from vk.com
-    vk_id = constants.vk_id
-    adr = 'https://api.vk.com/method/wall.get?owner_id={0}&count=1'.format(vk_id)
+def vk_get(vk_id):
+    """Get data from vk.com."""
+    adr = (
+        "https://api.vk.com/method" +
```
Comment (cleac) [inline vk.py:10]: BTW, what do you think if you move links to variables? Like `API_DOMAIN` in each file

**Pull request merged**